docs/root/configuration/http_filters/ext_authz_filter.rst

```diff
@@ -0,0 +1,52 @@
+.. _config_http_filters_ext_authz:
+
+External Authorization
+======================
+* External authorization service overview :ref:`architecture overview <arch_overview_ext_authz>`
+* :ref:`HTTP filter v2 API reference <envoy_api_msg_config.filter.http.ext_authz.v2alpha.ExtAuthz>`
+
+The external authorization HTTP filter calls an external gRPC service to check if the incoming HTTP request is authorized or not.
+If the request is deemed unauthorized then the request will be denied with 403 (Forbidden) response.
+
+.. tip::
+  It is recommended that this filter is configured first in the filter chain so that requests are authorized prior to rest of the filters running through the request.
+
+The content of the request that are passed to an authorization service is specified by :ref:`CheckRequest <envoy_api_msg_service.auth.v2alpha.CheckRequest>`
+
+.. _config_http_filters_ext_authz_http_configuration:
+
+The HTTP filter, using gRPC service, can be configured as follows. You can see all the configuration options at :ref:`HTTP filter <envoy_api_msg_config.filter.http.ext_authz.v2alpha.ExtAuthz>`
+
+Example
+-------
+
+A sample filter configuration could be:
+
+.. code-block:: yaml
+
+  http_filters:
+    - name: envoy.ext_authz
+      config:
+        grpc_service:
+           envoy_grpc:
+             cluster_name: ext-authz
+
+  clusters:
+    - name: ext-authz
+      type: static
+      http2_protocol_options: {}
+      hosts:
+        - socket_address: { address: 127.0.0.1, port_value: 10003 }
+
+Statistics
+----------
+The HTTP filter outputs statistics in the *cluster.<route target cluster>.ext_authz.* namespace.
```

Comment:
Are you sure this path is right? I suspect the stats are in the `http.<stat prefix>.` tree? Or was it done differently for this filter?

Reply:
Afaik this is the right path. see [here](https://github.com/envoyproxy/envoy/blob/26cd504c923b20d594ef4402717006275828d62b/source/extensions/filters/http/ext_authz/ext_authz.cc#L83)

Reply:
In general, I think this is odd, and you probably want to change this before we call this filter finished. The reason being that logically, you care more about what the filter is doing inline with the HTTP connection manager, network filter stack, versus the server that is doing the check (See for example what we do with rate limit stats). cc @gsagula